File: milk/src/test/java/com/yq/milk/Sub.java
package com.yq.milk;

/**
 * Created by king on 2018/2/23.
 */

public class Sub extends Super{
    String fName;

    public Sub(String fName, String lName ) {
        this.fName = fName;
        this.lName = lName;
        this.name = way();
    }

    public Sub(){}
    String lName;



    public String way(){
        return this.fName + this.lName;
    }
}

File: milk_kotlin/src/test/java/com/yq/milk_kotlin/Sub.kt
package com.yq.milk_kotlin

/**
 * Created by king on 2018/2/23.
 */
class Sub() : Super() {

    var fristName: String? = null
        get() = this.fristName

    var lastName: String? = null
    override lateinit var name: String

    init {
        name = add()
    }
    fun add(): String {
        return fristName + lastName
    }
//    override fun say(s: String): Super {
//        print("hello world")
//        return this
//    }
}